chipsec/hal/mmio.py

```diff
@@ -168,42 +152,48 @@ def is_MMIO_BAR_defined(self, bar_name):
             if self.logger.HAL: self.logger.warn( "'{}' MMIO BAR definition not found/correct in XML config".format(bar_name) )
         return is_bar_defined
 
+    #
+    # Get the base address and mask for a BAR register.
+    #
+    def _get_BAR_and_mask(self, bar_reg, base_field, bus):
```

Comment:
This routine needs some changes.  Line # 164 is looking for `bar` to determine `preserve` value.  Pulling that out and passing the result might work.

Reply:
Indeed I missed that when merging, and trusted the result because tests passed.

Do you have some documentation or example of using align_bits so that I can test my code? Pulling out seems to work for simple cases, but I'd like to put a bit more thought into the intended semantics if we have simultaneously register_high, base_field_high and align_bits fields (even though I can't see a use-case just yet since register_high is for AMD and align_bits for Intel so far).

Reply:
On second thought, I'm not sure it ever makes sense to have `preserve=True` for the high bits in case of split MMIO. It just so happens that all the examples I encountered for AMD had the high bits field starting at bit 0.

What do you think? If you don't feel strongly either way, I'll unconditionally set it to False (and maybe issue a warning if base_field_high is not starting at bit 0). I'm still interested in more details about align_bits semantics to check it plays well with register_high.

Reply:
Would it be easier to revert these changes and glean out the `register_high` routines?  Or maybe a MMIO sub-type help?  We'll take a look and possibly kick somethings around a bit to see what falls out.  I'm sure we can work out a better solution.
Also, it might help to break-out the specific AMD additions with the framework changes into separate PRs.  Might be easier to manage.  If we have to make changes to the framework part, it wouldn't interfere with the platform stuff.

Reply:
> Would it be easier to revert these changes and glean out the register_high routines? Or maybe a MMIO sub-type help?

I'm not sure what you have in mind, but please feel free to cherry-pick and refactor my commit if you have better ideas!

> Also, it might help to break-out the specific AMD additions with the framework changes into separate PRs. 

I tried to keep each commit self-contained, but I agree Github UI for reviews isn't well suited for such a pattern. I'd be happy to break things out in several PRs if it's easier for you. Mostly everything at this stage is touching the framework though, except the last commit which adds the XML config. There's not AMD-specific module or analysis.

Reply:
@frinzell would it help if we scheduled some time next week to discuss open issues (or even pair-program to quickly iron them out) over videoconference? I'm not sure about your timezone but feel free to reach out privately (firstname.lastname at ssi.gouv.fr).

Reply:
Hi @kerneis-anssi, Aaron is OOO this week. I am going to merge this in today but we should probably have a conversation around this next week if you have time. We are in the US Pacific/Central time zone.